app/src/main/java/com/omkarmoghe/pokemap/MainActivity.java

```diff
@@ -104,7 +106,15 @@ protected void onStop() {
 
     @Override
     public void onBackPressed() {
-        this.finish();
+        // As request, when the back button is pressed, we should pop the previous value
+        // (using immediate to execute immediate pop)
+        manager.popBackStackImmediate();
+
+        // Grab current amount of fragments in the stack, if 0, means that we can close the app
+        int backStackCount = manager.getBackStackEntryCount();
```

Comment:
wouldnt this close the app if there is one fragment on the stack? it pops, size is now 0 so it closes when we should only close if its 0 before poping?